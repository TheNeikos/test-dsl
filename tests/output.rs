#[test]
fn check_invalid() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
            }
            tetcase {
            }
            foobar {
            }
            asd
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_unknown() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat 2 {
                    not_found
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_missing_argument() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat {
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_argument_wrong_type() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat hello {
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}
