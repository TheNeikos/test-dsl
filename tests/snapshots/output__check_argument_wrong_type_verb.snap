---
source: tests/output.rs
expression: "format!(\"{:?}\", miette::Error::new(tc.unwrap_err()))"
---
  [31mÃ—[0m Testcase did not run successfully
[31m  â•°â”€â–¶ [0m  [31mÃ—[0m An argument was missing
[31m      [0m   â•­â”€[[36;1;4mtest.kdl:3:17[0m]
[31m      [0m [2m2[0m â”‚             testcase {
[31m      [0m [2m3[0m â”‚                 foobar
[31m      [0m   Â· [35;1m                â”€â”€â”€â”¬â”€â”€[0m
[31m      [0m   Â·                    [35;1mâ•°â”€â”€ [35;1mThis node is missing an argument[0m[0m
[31m      [0m [2m4[0m â”‚             }
[31m      [0m   â•°â”€â”€â”€â”€
[31m      [0m[36m  help: [0mThis function takes 1 arguments, you're missing the 1th argument.
[31m      [0m
